src/network/bpf/sysctl_monitor/sysctl-write-event.h

```diff
@@ -0,0 +1,42 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#pragma once
+
+#ifndef TASK_COMM_LEN
+#define TASK_COMM_LEN 16
+#endif
+
+/* It would be nice to size these members to bigger values, but the stack
+ * in BPF programs is limited to 512 bytes, and allocating bigger structures
+ * leads to this compile time error:
+ *   error: Looks like the BPF stack limit is exceeded.
+ *   Please move large on stack variables into BPF per-cpu array map.
+ *   For non-kernel uses, the stack can be increased using -mllvm -bpf-stack-size. */
+struct sysctl_write_event {
+        /* Used to track changes in the struct layout */
+        int version;
+
+        /* Error code returned to userspace to handle eventual failures. */
+        int errorcode;
+
+        /* The PID of the process which is writing the sysctl */
+        int pid;
+
```

Comment:
so, systemd loves cgroups. hence i think it would be great to pass the cgroup id along here too, right away, so that we don#t have to break protocol for this later.

i.e. use bpf_get_current_cgroup_id() on the bpf side.

(even if we don't make use of it just yet, though that would be cool too, though currently we have no helper for translating a cgroupid back to a path)

Reply:
Ok. To fit it in the stack I need to lower `current` and `newvalue` back to 160 bytes, but that's not a big deal.